src/main/java/com/solvd/shop24/gui/common/components/purchase/SearchProductItem.java

```diff
@@ -0,0 +1,40 @@
+package com.solvd.shop24.gui.common.components.purchase;
+
+import com.qaprosoft.carina.core.foundation.utils.factory.DeviceType;
+import com.qaprosoft.carina.core.foundation.utils.factory.ICustomTypePageFactory;
+import com.qaprosoft.carina.core.foundation.webdriver.decorator.ExtendedWebElement;
+import com.qaprosoft.carina.core.gui.AbstractUIObject;
+import com.solvd.shop24.gui.common.pages.purchase.ProductPageBase;
+import com.solvd.shop24.gui.desktop.pages.purchase.ProductPage;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+public class SearchProductItem extends AbstractUIObject implements ICustomTypePageFactory {
```

Comment:
in fact you don't initialize your AbstractUIObjects using ICustomTypePageFactory, you declared them as field in page class.
so you can remove implementation of ICustomTypePageFactory in this case.